### ISSUE TITLE
[Coq 8.9+] Please fix deprecation warnings in 8.9

### ISSUE THREAD

**Issue by ejgallego** [mit-plv/fiat#14]
Please fix deprecation warning in 8.9 so your plugin will compile with Coq 8.10, thanks!
```
CAMLOPT -c  src/Common/Tactics/transparent_abstract_tactics.ml
File "src/Common/Tactics/transparent_abstract_tactics.ml", line 22, characters 57-110:
Warning 3: deprecated: ArgVar
Use version in [Locus]
File "src/Common/Tactics/transparent_abstract_tactics.ml", line 31, characters 34-67:
Warning 3: deprecated: ArgVar
Use version in [Locus]
CAMLOPT -c  src/Common/Tactics/hint_db_extra_tactics.ml
File "src/Common/Tactics/hint_db_extra_tactics.ml", line 11, characters 58-111:
Warning 3: deprecated: ArgVar
Use version in [Locus]
File "src/Common/Tactics/hint_db_extra_tactics.ml", line 20, characters 32-65:
Warning 3: deprecated: ArgVar
Use version in [Locus]
CAMLC -c src/Common/Tactics/transparent_abstract_tactics.ml
File "src/Common/Tactics/transparent_abstract_tactics.ml", line 22, characters 57-110:
Warning 3: deprecated: ArgVar
Use version in [Locus]
File "src/Common/Tactics/transparent_abstract_tactics.ml", line 31, characters 34-67:
Warning 3: deprecated: ArgVar
Use version in [Locus]
```

**Comment by JasonGross** [comment]
Does 8.9 expose `ltac_apply`?  Then we can stop maintaining our own copy.

**Comment by ejgallego** [comment]
I have no idea @JasonGross , @ppedrot ?

**Comment by ejgallego** [comment]
Anyways I suggest you open a bug about `ltac_apply` upstream, and perform this cleanup which shouldn't be too hard.

**Comment by ppedrot** [comment]
There should be `Tacinterp.Value.apply` in 8.9.

**Comment by ejgallego** [comment]
Thanks @ppedrot

**Comment by ejgallego** [comment]
Ping.

**Comment by JasonGross** [comment]
I tried to do this with #15, but I'm getting
```
Error: Unbound value Tacinterp.Value.apply
```
with [71296e6e91de5b3fea08ef4b34426a02304f00b8](https://github.com/coq/coq/commits/71296e6e91de5b3fea08ef4b34426a02304f00b8)

What am I doing wrong?

**Comment by ejgallego** [comment]
Umm, c'est bizarre. I see nothing wrong _a priori_.

**Comment by ejgallego** [comment]
Well Travis passed on your own PR, so I dunno.

**Comment by JasonGross** [comment]
Well, `git clean -xfd` made it work locally, so I'll merge.

**Comment by ejgallego** [comment]
Thanks @JasonGross !

**Comment by ejgallego** [comment]
I cannot reopen an issue, but unfortunately we still have a problem:
```
File "src/Common/Tactics/hint_db_extra_tactics.ml", line 47, characters 62-86:
Error: Unbound record field Vernacexpr.hint_priority
```
Unfortunately OCaml doesn't allow to deprecate record fields :S so this must be manually replaced by `Typeclasses.hint_priority`

**Comment by JasonGross** [comment]
Fixed by c7285df856c1f79b8e2c220f8e86268f8842bdce

**Comment by ejgallego** [comment]
Thanks again @JasonGross , and sorry for the problem with the record field.
